sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -2803,19 +2804,23 @@ class AstBuilder(conf: SQLConf) extends SqlBaseBaseVisitor[AnyRef] with Logging
       Option(ctx.partitionSpec).map(visitNonOptionalPartitionSpec))
   }
 
+  override def visitColPosition(ctx: ColPositionContext): ColumnPosition = {
+    ctx.position.getType match {
+      case SqlBaseParser.FIRST => ColumnPosition.FIRST
+      case SqlBaseParser.AFTER => ColumnPosition.createAfter(ctx.afterCol.getText)
```

Comment:
shouldn't that be a visitIdentifier or something? You'd know better probably, but what would the parsing be with backticks, etc when you simply use `getText`?

Reply:
we only support `AFTER abc` not `AFTER 'abc'`. This is consistent with other places that need an identifier. e.g. `visitRenameTableColumn`

Reply:
backticks are excluded by the lexer, we don't need to worry about it here.

Reply:
I agree with @brkyvz. The identifier should be a multi-part identifier because it may be a nested column. This should visit the multi-part identifier to get a `Seq[String]`.

Reply:
Sorry, I think this is right after all. The parser uses a single identifier, so the syntax is actually `ADD COLUMN point.z AFTER y` not `ADD COLUMN point.z AFTER point.y`.

I think I'd prefer updating it to parse a multi-part identifier and accept both `point.y` and `y`, but that can be done as a follow-up.